### PR TITLE
feat(components/headerbar): uppercase on last name

### PR DESCRIPTION
**Pull request by ncomont**
**What is the problem this PR is trying to solve?**
The lastname is not in uppercase.

**What is the chosen solution to this problem?**
According to the guidelines, the lastname should be in uppercase ; so `lastName` and `firstName` have been added to the HeaderBar API. The existing behaviour is not altered by this PR.

**Please check if the PR fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 1]
```diff
@@ -153,20 +153,32 @@ Help.propTypes = {
...
 
+	let displayName = name;
+	if (firstName && lastName) {
+		displayName = (
+			<span className={classNames(theme['user-name'], 'user-name')}>
+				<span className={classNames(theme['user-firstname'], 'user-firstname')}>{firstName}</span>
+				<span className={classNames(theme['user-lastname'], 'user-firstname')}>{lastName}</span>
+			</span>
+		);
+	}
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Please move that `let` and if statement to a function.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 2]
```diff
@@ -153,20 +153,32 @@ Help.propTypes = {
...
+		theme['tc-header-bar-user'],
 		theme.separated,
 	);
 
+	let displayName = name;
+	if (firstName && lastName) {
+		displayName = (
+			<span className={classNames(theme['user-name'], 'user-name')}>
+				<span className={classNames(theme['user-firstname'], 'user-firstname')}>{firstName}</span>
+				<span className={classNames(theme['user-lastname'], 'user-firstname')}>{lastName}</span>
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Fix class name

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 3]
```diff
@@ -153,20 +153,32 @@ Help.propTypes = {
...
+
 	return (
 		<li className={className}>
 			<renderers.ActionDropdown
 				bsStyle="link"
 				icon="talend-user-circle"
 				pullRight
 				tooltipPlacement="bottom"
-				label={name}
+				tooltipLabel={name}
```
Comment (jare-talend) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Should you show displayName intead of name?

Comment (ncomont) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: THe only difference between `name` and  `displayName` is that `displayName` has `span` to apply the uppercase on the last name. For the `tooltipLabel`, we want the raw text.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 4]
```diff
@@ -153,21 +153,38 @@ Help.propTypes = {
...
+
+function User({ renderers, ...rest }) {
 	const className = classNames(
 		theme['tc-header-bar-action'],
+		theme['tc-header-bar-user'],
 		theme.separated,
 	);
 
+	function getDisplayName(params) {
+		const { name, firstName, lastName } = params;
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Please move that spread to the function arguments definition.

**Pull request merged**